packages/astro/src/runtime/client/visible.ts

```diff
@@ -11,8 +11,11 @@ const visibleDirective: ClientDirective = (load, options, el) => {
 		await hydrate();
 	};
 
-	const ioOptions = {
-		rootMargin: typeof options.value === 'string' ? options.value : undefined,
+	const rawOptions =
+		typeof options.value === 'object' ? (options.value as ClientVisibleOptions) : undefined;
```

Comment:
Just checking, do we already use `JSON.stringify`/`JSON.parse` on `options.value`? If not, we probably get a string here that needs to be deserialized.

Reply:
We do! I tested the change locally and everything works. I'm not sure where we do that, though. I'll add a E2E test.. somewhere